dev/Repeater/UniformGridLayoutState.cpp

```diff
@@ -172,7 +172,7 @@ void UniformGridLayoutState::ClearElementOnDataSourceChange(winrt::VirtualizingL
             break;
 
         case winrt::NotifyCollectionChangedAction::Move:
-            throw winrt::hresult_not_implemented();
+            shouldClear = args.NewStartingIndex() == 0 || args.OldStartingIndex() == 0;
```

Comment:
Should we add comments on why we need to check for those specific positions?

Reply:
I followed the pattern of the other cases which don't have any comments either, but I'm happy to add a comment if it doesn't seem clear enough. That said, all my understanding of this logic is based on reading the uncommented code, so I'm concerned about writing something which isn't completely accurate.

Reply:
I think the idea here, is that we check in what cases the event will have cleared the first item, because we are caching the first element. If any change changes the first item, we need to clear that element.

Reply:
+1 on adding some comments. UniformGridLayout's first element is special - we use the size of the first element to determine what the size of all the other elements are going to be. So if the first element changes, we need to re-evaluate everything.

Reply:
Done! Let me know if you have suggestions on the wording